core/src/main/java/c/org/rajawali3d/annotations/RequiresRenderTask.java

```diff
@@ -0,0 +1,20 @@
+package c.org.rajawali3d.annotations;
+
+import c.org.rajawali3d.control.RenderTask;
+import java.lang.annotation.Documented;
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * Flag a method that should run only in the context of a {@link RenderTask} or a similar
+ * context, i.e. on the RenderThread, and isolated from any other render events including frame renders
+ *
+ * @author Randy Picolet
+ */
+@Documented
+@Retention(RetentionPolicy.CLASS)
+@Target({ElementType.METHOD})
+public @interface RequiresRenderTask {
```

Comment:
Off the cuff, I am concerned this may be confusing without much aid. I will reserve final judgment until I get through the entirety of this PR. In the meantime, this comment will serve as a reminder to myself to re-evaluate.

Reply:
The point is to identify methods that should be called only in a render transaction. 

In my next update, I've renamed the `@RenderTask` annotation to `@RenderTransaction`, and this one to `@RequiresRenderTransaction`, as I co-opted the `RenderTask` class name from its old use in 1.x for start-of-frame `Renderer`-queued tasks to the new render-thread-queued tasks that run completely outside of draw frame events.  So in general, `@RenderTransaction` flags a client method that runs on the render thread but outside frame updates., so e.g. callbacks to clients running sequentially under a context creation, surface size change, or loss of context event control flow are `@RenderTransaction`s, in addition to the new `RenderTask` that allows apps to do whatever they need to when they need to.